### DISCUSSION TITLE
How do you use export --list?

### DISCUSSION THREAD

**Discussion by uhthomas** [cuelang/cue#800] [Q&A] [answered]
Hi, 

I'm just getting started with Cue and trying to get it working with Kubernetes. The Kubernetes API accepts either a list of delimited YAML files, or a list of JSON objects.

How can I export Cue to this format?

Thanks

**Comment by verdverm** [comment, marked as answer]
see `cue flags`

```
To handle multi-document files, such as JSON Lines or YAML
files with document separators (---), the user must specify
a the --path, --list, or --files flag.
The --path flag merges each element into a single package as
if each element was defined in a separate file. The --list flag
concatenates each entry in a file into a list.
Using --list flag in combination with the --path flag
concatenates entries with the same path into a list, instead of
unifying them.
Finally, the --files option causes each entry to be written to
a different file. The -files flag may only be used in
combination with the import command.
```

`--list` looks to work with data files that are merged with Cue, more on the input side of things.

---

For the output side, and specific to k8s, I have an example of what I think you may be trying here:

- https://github.com/hofstadter-io/hof-docs/blob/main/ci/cuelm.cue
- https://github.com/hofstadter-io/hof-docs/blob/main/ci/deploy.sh
- https://github.com/hofstadter-io/cuelm/blob/main/schema/common.cue

**Reply by uhthomas** [reply]
I see, it may have been an oversight on my part from forgetting that the `kubectl apply -f -` can accept a JSON document in the form:
```json
{
    "items": [{}, {}],
}
```

Thanks for the references, very useful!

---

However, I still don't really understand how `--list` is applicable to export.

**Reply by verdverm** [reply]
afaik, it is only applicable when you are combining Cue with json/yaml on the input / schema side of things, which can happen for `cue {def,eval,export,...}`. By default, cue is going to pick up on data files that are next to Cue files (depending on how you invoke Cue, it's complicated). There are flags that appear in several commands, some others are `-t`, `-l`, `-e`, `-d` and share their meaning between commands. Largely, the main commands share much of their processing or flow ( setting up and managing results the same ) and invoking Cue internals every so slightly differently.

**Reply by uhthomas** [reply]
I see, thanks for the detailed description. 😄 

I've got exactly what I want working using (k8s.io/api/core/v1).#List, so I'm happy.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/800.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
